runtime/onert/core/src/util/shapeinf/Range.cc

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+#include <cmath>
+namespace onert
+{
+namespace shape_inference
+{
+// helper function
+template <float *> ir::Shape inferRangeShape(float *start_val, float *limit_val, float *delta_val);
+template <typename T> ir::Shape inferRangeShape(T start_val, T limit_val, T delta_val)
```

Comment:
Please change this to `template <typename T> ir::Shape inferRangeShape(const T *start_val, const T *limit_val, const T *delta_val)`

Reply:
Changing the prototype of inferRangeShape to const T* from const T gives the following build errors in use of std::abs:

error: invalid conversion from ‘int*’ to ‘long long int’ [-fpermissive]
   30 |            ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(*delta_val))
      |                                                                                ~~~~~~~~^~~~~~~~~~~~
      |                                                                                        |
      |                                                                                        int*
I referred to a similar function GetSize under compute/cker/include/cker/operation/Range.h, and the template declaration is as follows: template <typename T> inline int GetSize(T start, T limit, T delta);

Reply:
| ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(**delta_val**)) 
-> 
| ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(***delta_val**))

Reply:
`is_integral<T>`'s `T` is not pointer, but you used params like pointer. You can use them uniformly

Reply:
> | ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(**delta_val**))
> ->
> | ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(***delta_val**))

I tried with both cases, I still see build errors similar to above.

> is_integral<T>'s T is not pointer, but you used params like pointer. You can use them uniformly

I understand your reasoning here. But the larger context of the errors lies in the use `std::abs`. With the use of T as pointer, as is implemented in `compute/cker/include/cker/operation/Range.h`, there are no build errors. Is it necessary to change this? Please let me know.